tests/test_admin.py

```diff
@@ -2408,3 +2408,119 @@ def test_change_view_action_compare_versions_three_selected(self):
+        self.assertContains(response, "cms-versioning-action-edit")
+        self.assertContains(response, "cms-versioning-action-manage-versions")
+        self.assertContains(response, "js-versioning-action")
+
+    def test_extended_version_change_get_list_display_without_cms_config(self):
+        """
+        Admin class should correctly add a default if cms_config is not configured
+        """
+
+        content_model = factories.BlogContentWithVersionFactory()
+        request = self.get_request("/")
+        blog_admin = admin.site._registry[BlogContent]
+
+        list_display = blog_admin.get_list_display(request=request)
+        list_actions_function = list_display[-1]
+
+        # List display should not be set
+        self.assertEqual(('__str__',), blog_admin.list_display)
+        # Check other values are populated
+        self.assertIn("get_author", list_display)
+        self.assertIn("get_modified_date", list_display)
+        self.assertIn("get_versioning_state", list_display)
+
+        # Fetching list_display through get_list_display should return extended list_display
+        list_actions = list_actions_function(content_model)
+
+        self.assertIn("cms-versioning-action-btn", list_actions)
+        self.assertIn("cms-versioning-action-preview", list_actions)
+        self.assertIn("cms-versioning-action-edit", list_actions)
+        self.assertIn("cms-versioning-action-manage-versions", list_actions)
+        self.assertIn("js-versioning-action", list_actions)
+
+
+class ListActionsTestCase(CMSTestCase):
+    def setUp(self):
+        self.modeladmin = admin.site._registry[PollContent]
+
+    def test_edit_link(self):
+        content_model = factories.BlogContentWithVersionFactory()
+        version = content_model.versions.last()
+        request = self.get_request("/")
+        request.user = self.get_superuser()
+        menu_content = version.content
+
+        func = self.modeladmin._list_actions(request)
+        edit_endpoint = reverse("admin:djangocms_versioning_pollcontentversion_edit_redirect", args=(version.pk,),)
+        response = func(menu_content)
+
+        self.assertIn("cms-versioning-action-btn", response)
+        self.assertIn('title="Edit"', response)
+        self.assertIn(edit_endpoint, response)
+
+    def test_edit_link_inactive(self):
+        content_model = factories.BlogContentWithVersionFactory()
+        version = content_model.versions.last()
+        request = self.get_request("/")
+        request.user = self.get_staff_user_with_no_permissions()
+
+        func = self.modeladmin._list_actions(request)
+        edit_endpoint = reverse("admin:djangocms_versioning_blogcontentversion_edit_redirect", args=(version.pk,),)
+        response = func(version.content)
+
+        self.assertIn("inactive", response)
+        self.assertIn('title="Edit"', response)
+        self.assertNotIn(edit_endpoint, response)
+
+    def test_edit_link_not_shown(self):
```

Comment:
```suggestion
    def test_edit_link_not_shown(self):
        """
        The edit link should not be shown when no user is logged in.
        """
```

Reply:
I'm not really sure what this test is doing because the user will have to be logged in to access the admin.